src/main/webapp/sources.js

```diff
@@ -78,8 +91,12 @@ export const Sources = props => {
 }
 
 const SourcesSelectComponent = props => {
-  const { sources = [], value = [], onChange } = props
-
+  const { sources = [], value = [], onChange, defaultValue } = props
+  useEffect(() => {
```

Comment:
Would it make sense to use the default value on every render where there is no value selected? It doesn't seem like there's much point in letting the user select nothing.

Reply:
My main reason was from a UX perspective. If my default is ['source1', 'source2', 'source3'], and I have that selected, then deselect all sources, as a user I would expect the sources to be empty (and maybe tell me that it will cause errors if left empty). I think it would be strange if after deselecting all three sources, you then have source1/2/3 selected.

Reply:
Maybe it would make sense to use the default value on every render where the value is undefined/null (the distinction being if it's an empty array, that would be allowed)?